website/source/docs/platform/k8s/injector/index.html.md

```diff
@@ -0,0 +1,264 @@
+The controller intercepts pod events and applies mutations to the pod if annotations exist within
+the request.  This functionality is provided by the [vault-k8s](https://github.com/hashicorp/vault-k8s)
+project and can be automatically installed and configured using the
+[Vault Helm](https://github.com/hashicorp/vault-helm) chart.
+
+## Overview
+
+The Vault Agent Injector works by intercepting pod `CREATE` and `UPDATE`
+events in Kubernetes.  The controller parses the event and looks for the metadata
+annotation `vault.hashicorp.com/agent-inject: true`.  If found, the controller will
+alter the pod specification based on other annotations present.
+
+### Mutations
+
+At a minimum, every container in the pod will be configured to mount a shared
+memory volume.  This volume, mounted to `/vault/secrets`, will be used by the Vault
+Agent containers for sharing secrets with the other containers in the pod.
+
+Next, two types of Vault Agent containers  can be injected: init and sidecar.  The
+init container will prepopulate the shared memory volume with the requested
+secrets prior to the other containers starting.  The sidecar container will
+continue to authenticate and render secrets to the same location as the pod runs.
+Using annotations, the initialization and sidecar containers may be disabled.
+
+Last, two additional types of volumes can be optionally mounted to the Vault Agent
+containers.  The first is secret volume containing TLS requirements such as client
+and CA (certificate authority) certificates and keys. This volume is useful when
+communicating and verifying the Vault servers authenticity using TLS.  The second
+is a configuration map containing Vault Agent configuration files.  This volume is
+useful to customize Vault Agent beyond what the provided annotations offer.
+
+### Authenticating with Vault
+
+The primary method of authentication with Vault when using the Vault Agent Injector
+is the service account attached to the pod.  At this time, no other authentication
+method is supported by the controller.
+
+The service account must be bound to a Vault role and a policy granting access to
+the secrets desired.
+
+A service account must be present to use the Vault Agent Injector.  It is *not*
+recommended to bind Vault roles to the default service account provided to pods
+if no service account is defined.
+
+### Requesting Secrets
+
+There are two methods of configuring the Vault Agent containers to render secrets:
+
+* the `vault.hashicorp.com/agent-inject-secret` annotation, or
+* a configuration map containing Vault Agent configuration files.
+
+Only one of these methods may be used at any time.
+
+#### Secrets via Annotations
+
+To configure secret injection using annotations, the user must supply:
+
+* one or more _secret_ annotations, and
+* the Vault role used to access those secrets. 
+
+The annotation must have the format:
+
+```yaml
+vault.hashicorp.com/agent-inject-secret-<unique-name>: /path/to/secret
```

Comment:
What are the constraints on `<unique-name>`? Can I make it `foo/bar` and get secret "bar" in directory "foo"?

Reply:
No, that can't be done.  That's actually not a valid annotation per Kube:

```
Invalid value: "vault.hashicorp.com/agent-inject-secret-db/creds": a qualified name must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]') with an optional DNS subdomain prefix and '/' (e.g. 'example.com/MyName')
```

I'll add a note about this.

Reply:
Fixed!